### ISSUE TITLE
windows-2022 version 20241113.3.0 breaks our build

### ISSUE THREAD

**Issue by Bouke** [actions/runner-images#10987]
### Description

We're using [repeatable package restores](https://devblogs.microsoft.com/nuget/enable-repeatable-package-restores-using-a-lock-file/) in our .NET build pipeline. However as version 20241113.3.0 started rolling out to agents, our build has been failing:

> error NU1004: The project's runtime identifiers have changed from. Project's runtime identifiers: win-x86, lock file's runtime identifiers win7-x86.The packages lock file is inconsistent with the project dependencies so restore can't be run in locked mode. Disable the RestoreLockedMode MSBuild property or pass an explicit --force-evaluate option to run restore to update the lock file.

We're still targeting .NET 8, not .NET 9, so I don't understand why this is now failing. We've since tried updating the lockfiles. However as some agents are still running 20241021.1.0, the updated lockfiles will fail on _those_. This is not a great situation to be in. Also updating the lockfiles requires all our developers to install .NET 9 SDK, which is something we would like to avoid.

### Platforms affected

- [X] Azure DevOps
- [ ] GitHub Actions - Standard Runners
- [ ] GitHub Actions - Larger Runners

### Runner images affected

- [ ] Ubuntu 20.04
- [ ] Ubuntu 22.04
- [ ] Ubuntu 24.04
- [ ] macOS 12
- [ ] macOS 13
- [ ] macOS 13 Arm64
- [ ] macOS 14
- [ ] macOS 14 Arm64
- [ ] macOS 15
- [ ] macOS 15 Arm64
- [ ] Windows Server 2019
- [X] Windows Server 2022

### Image version and build link

20241113.3.0

### Is it regression?

20241021.1.0

### Expected behavior

Projects targeting .NET 8 (LTS) should continue to do so.

### Actual behavior

Build fails.

### Repro steps

-

**Comment by vidyasagarnimmagaddi** [comment]
Hi @Bouke , Thank you for bringing this issue. we will revert once we have an update. thanks

**Comment by Bouke** [comment]
Hi @vidyasagarnimmagaddi, any update on this?

**Comment by jonagh** [comment]
Our project(s) are also suddenly failing dotnet build on windows-latest runner.

We have a bunch of warnings suddenly (at least some of which seem to have been added for projects that target .NET 9). Note that we are NOT targeting .NET 9.

We treat warnings as errors in CI/CD build so these new warnings fail the build. I guess we'll have to change this setting to work around the issue, so we can continue building/deploying - but it's not ideal.

Yes, we should (and hopefully will eventually) fix the warnings but these are legacy projects which will take time to update (and not break along the way), and it's not a priority - they have been running fine for a long time. Now in order to make this work we have to ignore these warnings which were not there a moment ago 😒

Note that I have tested running previously-succeeding code through our build-workflow, and as expected it fails (since it is now using the new runner image).

The only difference I see is the runner image version, which is same as mentioned by @Bouke.
- 20241021.1.0 was working
- 20241113.3.0 new runner image causes our build to fail (due to new warnings re .NET 9 which we are not targeting)

**Comment by vidyasagarnimmagaddi** [comment]
Hi @Bouke ,We observed that some agents are running with `20241021.1.0`
 We plan to start  image rollout this week,once this is done, issue will be resolved.

**Comment by Bouke** [comment]
@vidyasagarnimmagaddi  What do you mean with this, will you revert the rollout of 20241113.3.0?

**Comment by vidyasagarnimmagaddi** [comment]
> @vidyasagarnimmagaddi What do you mean with this, will you revert the rollout of 20241113.3.0?

Hi @Bouke , i mean new image rollout .

**Comment by Bouke** [comment]
Can you be more specific in how you're going to address this issue? For example: "we're going to complete the 20241113.3.0 image rollout, and you'll not see 20241021.1.0 again". If that's the case: that doesn't solve the issue: we'd still need all our developers to install .NET 9 SDK.

**Comment by vidyasagarnimmagaddi** [comment]
Hi @Bouke  , When we rollout new  image version this week, there will not be any agents with older version i.e, (`20241021.1.0`, `20241113.3.0`) ,. We make sure all the agents are with the newly released version only. thanks.

**Comment by Bouke** [comment]
Please consider rolling back .NET 9 SDK. There's a known issue with .NET 9 that silently breaks our app in production, which is not going to be fixed until .NET 10 SDK: https://github.com/dotnet/sdk/pull/44160#issuecomment-2485805964.

**Comment by normeno** [comment]
I have a similar problem. I use MSBuild and with version 20241021.1.0 it works, but with version 20241113.3.0, my workflow breaks

20241021.1.0:
![image](https://github.com/user-attachments/assets/d715a3fb-7a83-4f3d-aef5-af8f6f157c0d)

20241113.3.0:
![image](https://github.com/user-attachments/assets/fb27a0ea-3731-458a-a17a-bb184f07c827)

Unfortunately that version is not something I can select and I don't know how to fix it.

**Comment by vidyasagarnimmagaddi** [comment]
Hi @Bouke , we have re-deployed all the agents to `20241113.3.0` , kindly try to build . And revert if any error. thanks

**Comment by Bouke** [comment]
@vidyasagarnimmagaddi did you read anything of what I posted?

**Comment by vidyasagarnimmagaddi** [comment]
> Please consider rolling back .NET 9 SDK. There's a known issue with .NET 9 that silently breaks our app in production, which is not going to be fixed until .NET 10 SDK: [dotnet/sdk#44160 (comment)](https://github.com/dotnet/sdk/pull/44160#issuecomment-2485805964).

Hi @Bouke  , we  are currently anayzing the and will revert you . thanks

**Comment by dvdgzl** [comment]
@vidyasagarnimmagaddi Is there any update on this? 20241113.3.0 has broken our builds as well which may cause a delay in an upcoming release. Randomly, we had one successful build that somehow used an unreleased version 20241119.5.0. If we could keep using that version it would be great :) Thank you!

![image](https://github.com/user-attachments/assets/a7c2091f-2f5c-46ac-9b0d-836216557ef1)

**Comment by vidyasagarnimmagaddi** [comment]
Hi @Bouke , @dvdgzl  
The new image with version : `20241125.1` has been rolled out . please try to  build and revert if any issues. thanks

**Comment by vidyasagarnimmagaddi** [comment]
Hi @Bouke , @dvdgzl
The new image with version : 20241125.1 has been rolled out . please try to build and revert if any issues. thanks

**Comment by vidyasagarnimmagaddi** [comment]
Hi @Bouke , @dvdgzl
The new image with version : 20241125.1 has been rolled out . please try to build and revert if any issues. thanks
